artifishswarm/afsa.py

```diff
@@ -84,12 +90,13 @@ def prey(self, fish_idx: int):
         :return: None
         """
         for _ in range(self.search_retries):
-            target_x = self.fish[fish_idx] + self.vision * self.rng.uniform(-1, 1)
+            target_x = self.fish[fish_idx] + \
+                self.vision * self.rng.uniform(-1, 1)
             # originally the random is between [0,1) but it may be worth exploring
             # how it behaves when allowed the range [-1,1]
 
             target_y = self.func(target_x)
-            if target_y > self.func(self.fish[fish_idx]):
+            if self.comparator(target_y, self.func(self.fish[fish_idx])):
```

Comment:
Normally, comparator returns a value > 0 if the first argument is greater than the second, < 0 if the first is less than the second, and 0 if they are equal. I would suggest applying these semantics here.